src/Components/App/App.js

```diff
@@ -38,37 +40,46 @@ const App = () => {
 
   return (
     <section className="App">
-      <Header />
-      <Switch>
-        <Route path="/about" render={() => <About />} />
-        <Route path="/styles" render={() => <Styles />} />
-        <Route path="/contact" render={() => <ContactForm />} />
-        <Route
-          path="/stylist/:id"
-          render={({ match }) => {
-            const { id } = match.params;
-            return <StylistPage id={id} />;
-          }}
-        />
-        <Route
-          exact
-          path="/"
-          render={() => (
-            <>
-              <SearchForm
-                retrieveSearchResults={retrieveSearchResults}
-                setIsLoading={setIsLoading}
-              />
-              <SearchResultsContainer
-                results={results}
-                style={style}
-                isLoading={isLoading}
-              />
-            </>
-          )}
-        ></Route>
-      </Switch>
-      <Footer />
+      {!isModalOpen && (
+        <React.Fragment>
+          <Header setIsModalOpen={setIsModalOpen} />
+          <Switch>
+            <Route path="/about" render={() => <About />} />
+            <Route path="/styles" render={() => <Styles />} />
+            <Route path="/contact" render={() => <ContactForm />} />
+            <Route
+              path="/stylist/:id"
+              render={({ match }) => {
+                const { id } = match.params;
+                return <StylistPage id={id} />;
+              }}
+            />
+            <Route
+              exact
+              path="/"
+              render={() => (
+                <>
+                  <SearchForm
+                    retrieveSearchResults={retrieveSearchResults}
+                    setIsLoading={setIsLoading}
+                  />
+                  <SearchResultsContainer
+                    results={results}
+                    style={style}
+                    isLoading={isLoading}
+                  />
+                </>
+              )}
+            ></Route>
+          </Switch>
+          <Footer />
+        </React.Fragment>
+      )}
+      {isModalOpen && (
```

Comment:
If the modal is open, does not display the header and footer and page contents and instead displays the modal only.